### PR TITLE
asciiquarium: fix test logic

### PR DESCRIPTION
**Pull request by iMichka**
- The stdin/sdtout variable names were inverted
- Fixes: Errno::EIO: Input/output error @ io_fread - /dev/pts/2 on linux

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
@homebrew/core I need help with PTY.spawn, does anyone have experience with this? It's not working correctly on 11.0 and 10.15. Can someone test the output there?

**Comment by carlocab** [comment]
I think it's broken:
```
❯ asciiquarium
[1]    22509 segmentation fault  asciiquarium
```

**Comment by carlocab** [comment]
`asciiquarium` is an env wrapper for a perl script. So here's what I tried:
```
❯ perl /usr/local/Cellar/asciiquarium/1.1_1/libexec/bin/asciiquarium
Can't locate Term/Animation.pm in @INC (you may need to install the Term::Animation module) (@INC contains: /usr/local/Cellar/perl/5.32.1/lib/perl5/site_perl/5.32.1/darwin-thread-multi-2level /usr/local/Cellar/perl/5.32.1/lib/perl5/site_perl/5.32.1 /usr/local/Cellar/perl/5.32.1/lib/perl5/5.32.1/darwin-thread-multi-2level /usr/local/Cellar/perl/5.32.1/lib/perl5/5.32.1 /usr/local/lib/perl5/site_perl/5.32.1) at /usr/local/Cellar/asciiquarium/1.1_1/libexec/bin/asciiquarium line 42.
BEGIN failed--compilation aborted at /usr/local/Cellar/asciiquarium/1.1_1/libexec/bin/asciiquarium line 42.
```
That said, I don't know perl, so that may not have even been the right thing to do.

**Comment by iMichka** [comment]
I think I know what to do, I need to vendor that library. W do that with other perl formulae. Let me try that.

**Comment by carlocab** [comment]
If I stick it in `lldb`:
```
❯ lldb /usr/bin/perl
(lldb) target create "/usr/bin/perl"
Current executable set to '/usr/bin/perl' (x86_64).
(lldb) env PERL5LIB="/usr/local/Cellar/asciiquarium/1.1_1/libexec/lib/perl5"
(lldb) run /usr/local/Cellar/asciiquarium/1.1_1/libexec/bin/asciiquarium
error: process exited with status -1 (attach failed (Not allowed to attach to process.  Look in the console messages (Console.app), near the debugserver entries when the attached failed.  The subsystem that denied the attach permission will likely have logged an informative message about why it was denied.))
```

Console log: 
[crash.tar.gz](https://github.com/Homebrew/homebrew-core/files/5961441/crash.tar.gz)

**Comment by mitchblank** [comment]
@iMichka / @carlocab -- you might take a look at my previous attempt to fix this formula: #66485

I had it working on my 11.0 box but had issues getting CI to pass.  Ultimately the PR never got merged.  Maybe you fixed the test issue now and just need to integrate those other fixes?

**Comment by mitchblank** [comment]
It's a bit unfortunate since the package itself seems to build/run/test fine locally for me (at least after the changes I made) .  Shame that the CI system has some problem with some formulae these days...

Or are you seeing the same failure that @carlocab was?  On which OS?

**Comment by nandahkrishna** [comment]
@iMichka could you check if this is alright? Just took a shot and it seems to have worked.

**Comment by carlocab** [comment]
I think @iMichka will be happy to take any fix here. 😄

File: Formula/asciiquarium.rb [hunk 1]
```diff
@@ -46,7 +47,7 @@ def install
     # Disable dynamic selection of perl which may cause segfault when an
     # incompatible perl is picked up.
     # https://github.com/Homebrew/homebrew-core/issues/4936
-    inreplace "asciiquarium", "#!/usr/bin/env perl", "#!/usr/bin/perl"
+    inreplace "asciiquarium", "#!/usr/bin/env perl", "#!#{Formula["perl"].opt_bin/"perl"}"
```
Comment (carlocab) [inline Formula/asciiquarium.rb:50]: Can we use a `shebang` replacement method here?

File: Formula/asciiquarium.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class Asciiquarium < Formula
   homepage "https://robobunny.com/projects/asciiquarium/html/"
   url "https://robobunny.com/projects/asciiquarium/asciiquarium_1.1.tar.gz"
   sha256 "1b08c6613525e75e87546f4e8984ab3b33f1e922080268c749f1777d56c9d361"
-  license "GPL-2.0"
+  license "GPL-2.0-or-later"
   revision 1
```
Comment (carlocab) [inline Formula/asciiquarium.rb:7]: We should probably bump the revision for the dependency change.

**Review by SMillerDev — APPROVED** [review]

**Comment by iMichka** [comment]
Nice work. So it just missed a "q" to quit the process. People will be happy to get the fishes back on their screens :)

**Pull request closed without merging**